lib/smart_proxy_host_reports/ansible_processor.rb

```diff
@@ -132,4 +132,13 @@ def process_message(result)
   ensure
     result["friendly_message"] = msg
   end
+
+  def process_statuses
+    {
+      applied: @body["status"]["applied"],
+      failed: @body["status"]["failed"],
+      pending: @body["status"]["pending"] || 0, # It's only present in check mode
+      other: @body["status"]["skipped"],
+    }
+  end
```

Comment:
I suggest to add a rescue block and return zeroes in case anything goes wrong, drop a warning into logs as well. These keys should be always present but for consistency with Puppet.